contracts/src/FundAgent.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.20;
+
+import {WETH9} from "canonical-weth/WETH9.sol";
+import {Script} from "forge-std/Script.sol";
+import {BeefyClient} from "./BeefyClient.sol";
+
+import {IGateway} from "./interfaces/IGateway.sol";
+import {GatewayProxy} from "./GatewayProxy.sol";
+import {Gateway} from "./Gateway.sol";
+import {GatewayUpgradeMock} from "../test/mocks/GatewayUpgradeMock.sol";
+import {Agent} from "./Agent.sol";
+import {AgentExecutor} from "./AgentExecutor.sol";
+import {ParaID, Config} from "./Types.sol";
+import {SafeNativeTransfer} from "./utils/SafeTransfer.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
+contract FundAgent is Script {
+    using SafeNativeTransfer for address payable;
+    using stdJson for string;
+
+    function setUp() public {}
+
+    function run() public {
+        uint256 privateKey = vm.envUint("PRIVATE_KEY");
+        address deployer = vm.rememberKey(privateKey);
+        vm.startBroadcast(deployer);
+
+        uint256 initialDeposit = vm.envUint("BRIDGE_HUB_INITIAL_DEPOSIT");
```

Comment:
I see. I just think we need to think about the maintenance overhead that these scripts add. Is it not better to do the necessary funding in the test itself?

Reply:
Yes, agree. It can be executed directly with `forge script` as we do for deploying contracts and we can add a thin shell wrapper for it. 

But I would prefer to improve these maintenance scripts in a separate PR and make this one more focused. As you mentioned it's already a mammoth PR now😄 🐘.

Reply:
https://github.com/Snowfork/snowbridge/pull/940/commits/9a905a4eb3ea09d010d0a95877a4876311cc18d2